src/Container.ts

```diff
@@ -162,8 +182,45 @@ export class Container {
     /**
      * Provides a set of values to be saved in the container.
      */
-    static provide(values: { id: ServiceIdentifier, value: any }[]) {
-        values.forEach((v: any) => this.set(v.id, v.value));
+    static provide(providers: (Provider | ObjectType<any>)[]) {
```

Comment:
idea of "provide" was to make it work same as "set", but for multiple items. But right now "set" does not support everything provide supports. Maybe move some peaces of code into "set" method and here simply call it?

Reply:
also we can remove `registerHandler` function and merge its functionality into `set` method:

```ts

    /**
     * Sets a value for the given type or service name in the container.
     */
    static set<T, K extends keyof T>(service: ServiceMetadata<T, K>): Container;

    /**
     * Sets a value for the given type or service name in the container.
     */
    static set(type: Function, value: any): Container;

    /**
     * Sets a value for the given type or service name in the container.
     */
    static set(name: string, value: any): Container;

    /**
     * Sets a value for the given type or service name in the container.
     */
    static set(token: Token<any>, value: any): Container;

    /**
     * Sets a value for the given type or service name in the container.
     */
    static set(identifierOrServiceMetadata: ServiceIdentifier|ServiceMetadata<any, any>, value?: any): Container {

        const newService: ServiceMetadata<any, any> = arguments.length === 1 && identifierOrServiceMetadata instanceof Object ? identifierOrServiceMetadata : undefined;
        if (newService) {
            const service = this.findService(newService.id);
            if (service) {
                Object.assign(service, newService);
            } else {
                this.services.push(newService);
            }

        } else {
            const identifier = identifierOrServiceMetadata as ServiceIdentifier;
            const service = this.findService(identifier);
            if (service) {
                service.instance = value;

            } else {
                const service: ServiceMetadata<any, any> = {
                    instance: value
                };
                if (identifier instanceof Token || typeof identifier === "string") {
                    service.id = identifier;

                } else {
                    service.type = identifier;
                }

                this.services.push(service);
            }
        }

        return this;
    }
```